packages/next/next-server/server/config.ts

```diff
@@ -10,7 +10,60 @@ import * as Log from '../../build/output/log'
 const targets = ['server', 'serverless', 'experimental-serverless-trace']
 const reactModes = ['legacy', 'blocking', 'concurrent']
 
-const defaultConfig: { [key: string]: any } = {
+interface Config {
+  env: any[]
+  webpack: any | null
```

Comment:
Are the nulls necessary?
```suggestion
  webpack: (config: webpack.Configuration, { dir: string, dev: boolean, isServer: boolean, buildId: string }) => webpack.Configuration | Promise<webpack.Configuration>
```